File: solutions/python/raindrops/1/raindrops.py
def convert(number):
    result = ""
    
    if number % 3 == 0:
        result += "Pling"
    if number % 5 == 0:
        result += "Plang"
    if number % 7 == 0:
        result += "Plong"
    
    # If no factors were found, return the number as string
    if result == "":
        result = str(number)
    
    return result

File: solutions/c/raindrops/1/raindrops.c
// raindrops.c
#include "raindrops.h"
#include <stdio.h>
#include <string.h>

void convert(char result[], int drops) {
    // Start with empty string
    result[0] = '\0';
    
    // Check divisibility and append corresponding strings
    if (drops % 3 == 0) {
        strcat(result, "Pling");
    }
    if (drops % 5 == 0) {
        strcat(result, "Plang");
    }
    if (drops % 7 == 0) {
        strcat(result, "Plong");
    }
    
    // If no sounds were added (result is still empty), convert number to string
    if (result[0] == '\0') {
        sprintf(result, "%d", drops);
    }
}